modules/sdk/gradle-plugins-wsdd-builder/README.markdown

```diff
@@ -0,0 +1,117 @@
+(by default: `src/main/resources`).
+
+Instead, if the [`war`](https://docs.gradle.org/current/userguide/war_plugin.html)
+plugin is applied, the task uses `${project.webAppDir}/WEB-INF/service.xml` as
+input, and generates `${project.webAppDir}/WEB-INF/server-config.wsdd`. The
+`*_deploy.wsdd` and `*_undeploy.wsdd` files are still generated in the first
+`resources` directory of the `main` source set.
+
+Liferay WSDD Build Service requires an additional classpath (configured with the
+`buildWSDD.builderClasspath` property) in order to correctly generate the WSDD
+files. The `buildWSDD` task uses the following default value, which creates an
+implicit dependency to the `compileJava` task:
+
+```gradle
+tasks.compileJava.outputs.files + sourceSets.main.compileClasspath + sourceSets.main.runtimeClasspath
+```
+
+### BuildWSDDTask
+
+Tasks of type `BuildWSDDTask` extend [`JavaExec`](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.JavaExec.html),
+so all its properties and methods, like [`args`](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.JavaExec.html#org.gradle.api.tasks.JavaExec:args(java.lang.Iterable))
+and [`maxHeapSize`](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.JavaExec.html#org.gradle.api.tasks.JavaExec:maxHeapSize)
+are available. They also have the following properties set by default:
+
+Property Name | Default Value
+------------- | -------------
+[`args`](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.JavaExec.html#org.gradle.api.tasks.JavaExec:args) | WSDD Builder command line arguments
+[`classpath`](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.JavaExec.html#org.gradle.api.tasks.JavaExec:classpath) | [`project.configurations.wsddBuilder`](#liferay-wsdd-builder-dependency)
+[`main`](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.JavaExec.html#org.gradle.api.tasks.JavaExec:main) | `"com.liferay.portal.tools.wsdd.builder.WSDDBuilder"`
+
+#### Task Properties
+
+Property Name | Type | Default Value | Description
+------------- | ---- | ------------- | -----------
+`builderClasspath` | `String` | `null` | The classpath used by Liferay WSDD Builder to generate the WSDD files.
+`inputFile` | `File` | `null` | The `service.xml` from which to generate the WSDD files.
+`outputDir` | `File` | `null` | The directory where the `*_deploy.wsdd` and `*_undeploy.wsdd` files should be generated.
+`serverConfigFile` | `File` | `${project.projectDir}/server-config.wsdd` | The destination of the generated `server-config.wsdd` file.
+`serviceNamespace` | `String` | `"Plugin"` | The WSDD Service namespace.
+
+The properties of type `File` supports any type that can be resolved by [`project.file`](https://docs.gradle.org/current/dsl/org.gradle.api.Project.html#org.gradle.api.Project:file(java.lang.Object)).
+Moreover, it is possible to use Closures and Callables as values for the
+`String` properties, in order to defer their evaluation at the moment of the
+task execution.
```

Comment:
Great. Thanks for the quick responses, Andrea!


On 7/8/2016 12:25 PM, Andrea Di Giorgi wrote:
>
> In modules/sdk/gradle-plugins-wsdd-builder/README.markdown
> <https://github.com/jhinkey/liferay-portal/pull/367#discussion_r70101511>:
>
> > +[`main`](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.JavaExec.html#org.gradle.api.tasks.JavaExec:main) | `"com.liferay.portal.tools.wsdd.builder.WSDDBuilder"`
> > +
> > +#### Task Properties
> > +
> > +Property Name | Type | Default Value | Description
> > +------------- | ---- | ------------- | -----------
> > +`builderClasspath` | `String` | `null` | The classpath used by Liferay WSDD Builder to generate the WSDD files.
> > +`inputFile` | `File` | `null` | The `service.xml` from which to generate the WSDD files.
> > +`outputDir` | `File` | `null` | The directory where the `*_deploy.wsdd` and `*_undeploy.wsdd` files should be generated.
> > +`serverConfigFile` | `File` | `${project.projectDir}/server-config.wsdd` | The destination of the generated `server-config.wsdd` file.
> > +`serviceNamespace` | `String` | `"Plugin"` | The WSDD Service namespace.
> > +
> > +The properties of type `File` supports any type that can be resolved by [`project.file`](https://docs.gradle.org/current/dsl/org.gradle.api.Project.html#org.gradle.api.Project:file(java.lang.Object)).
> > +Moreover, it is possible to use Closures and Callables as values for the
> > +`String` properties, in order to defer their evaluation at the moment of the
> > +task execution.
>
> I think the best way of saying it is "defer evaluation until task
> execution".
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/jhinkey/liferay-portal/pull/367/files/a980814d392c8824bc1230fa56ead658e7eedf0b#r70101511>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AArapX2wLhOKSN0-OM03P79KotE5HZ2xks5qTnnegaJpZM4I8VXz>.
>

-- 
Jim Hinkey
Sr. Knowledge Engineer
Liferay, Inc.
www.liferay.com <http://www.liferay.com>

Visit Us: www.liferay.com <http://www.liferay.com> | Like Us:
facebook.com/liferay <http://facebook.com/liferay> | Follow Us:
twitter.com/liferay <http://twitter.com/liferay>